### PR TITLE
parser: add SHOW CREATE USER syntax

### PR DESCRIPTION
**Pull request by lnhote**
### What problem does this PR solve? 
parser PR: https://github.com/pingcap/parser/pull/156 
parser issue: https://github.com/pingcap/parser/issues/137

### What is changed and how it works?
1. add a new type for ShowStmt
2. update parser.y and parser.go

### Check List 

Tests 

 - Unit test

Code changes
 - add new const: ast.ShowCreateUser

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8970)

### CODE REVIEW

**Comment by zz-jason** [comment]
@lnhote https://github.com/pingcap/parser/pull/156 is merged, could you update this PR to point to the latest "pingcap/parser" branch so that we can carry on this PR?

**Comment by zz-jason** [comment]
LGTM

File: go.sum [hunk 1]
```diff
@@ -81,6 +81,12 @@ github.com/klauspost/cpuid v0.0.0-20170728055534-ae7887de9fa5 h1:2U0HzY8BJ8hVwDK
 github.com/klauspost/cpuid v0.0.0-20170728055534-ae7887de9fa5/go.mod h1:Pj4uuM528wm8OyEC2QMXAi2YiTZ96dNQPGgoMS4s3ek=
 github.com/konsorten/go-windows-terminal-sequences v1.0.1 h1:mweAR1A6xJ3oS2pRaGiHgQ4OO8tzTaLawm8vnODuwDk=
 github.com/konsorten/go-windows-terminal-sequences v1.0.1/go.mod h1:T0+1ngSBFLxvqU3pZ+m/2kptfBszLMUkC4ZK/EgS/cQ=
+github.com/mattn/go-colorable v0.0.9 h1:UVL0vNpWh04HeJXV0KLcaT7r06gOH2l4OW6ddYRUIY4=
+github.com/mattn/go-colorable v0.0.9/go.mod h1:9vuHe8Xs5qXnSaW/c/ABM9alt+Vo+STaOChaDxuIBZU=
+github.com/mattn/go-isatty v0.0.4 h1:bnP0vzxcAdeI1zdubAl5PjU6zsERjGZb7raWodagDYs=
+github.com/mattn/go-isatty v0.0.4/go.mod h1:M+lRXTBqGeGNdLjl/ufCoiOlB5xdOkqRJdNxMWT7Zi4=
+github.com/mattn/go-runewidth v0.0.3 h1:a+kO+98RDGEfo6asOGMmpodZq4FNtnGP54yps8BzLR4=
+github.com/mattn/go-runewidth v0.0.3/go.mod h1:LwmH8dsx7+W8Uxz3IHJYH5QSwggIsqBzpuz5H//U1FU=
```
Comment (lamxTyler) [inline go.sum]: Why add these lines?

**Pull request merged**